pkg/node/node.go

```diff
@@ -474,34 +507,126 @@ func (n *Node) monitorKeepPublicKeySubmission(
 	}
 
 	defer subscriptionConflictingPublicKey.Unsubscribe()
-	defer subscriptionPublicKeyPublished.Unsubscribe()
 
-	select {
-	case event := <-publicKeyPublished:
-		logger.Infof(
-			"public key [%x] has been accepted by keep: [%s]",
-			event.PublicKey,
-			keepAddress.String(),
-		)
-	case event := <-conflictingPublicKey:
+	startingBlock, err := n.ethereumChain.BlockCounter().CurrentBlock()
+	if err != nil {
 		logger.Errorf(
-			"member [%x] has submitted conflicting public key for keep [%s]: [%x]",
-			event.SubmittingMember,
-			keepAddress.String(),
-			event.ConflictingPublicKey,
-		)
-	case <-monitoringCtx.Done():
-		logger.Warningf(
-			"monitoring of public key submission for keep [%s] "+
-				"has been cancelled: [%v]",
+			"failed to get the starting block while setting up "+
+				"public key submission monitoring for keep [%s]: [%v]",
 			keepAddress.String(),
-			monitoringCtx.Err(),
+			err,
 		)
-	case <-abort:
-		logger.Warningf(
-			"monitoring of public key submission for keep [%s] "+
-				"has been aborted",
+		return
+	}
+
+	pubkeyCheckTrigger, err := n.ethereumChain.BlockCounter().BlockHeightWaiter(
+		startingBlock + blockConfirmations,
+	)
+	if err != nil {
+		logger.Errorf(
+			"failed to get the check trigger while setting up "+
+				"public key submission monitoring for keep [%s]: [%v]",
 			keepAddress.String(),
+			err,
 		)
+		return
+	}
+
+	for {
+		select {
+		case event := <-conflictingPublicKey:
+			logger.Errorf(
+				"member [%x] has submitted conflicting public key for keep [%s]: [%x]",
+				event.SubmittingMember,
+				keepAddress.String(),
+				event.ConflictingPublicKey,
+			)
+			return
+		case pubkeyCheckBlock := <-pubkeyCheckTrigger:
```

Comment:
Let's explain why do we execute the check periodically instead of waiting for an event.